funding/batch.go

```diff
@@ -230,19 +230,31 @@ func (b *Batcher) BatchFund(ctx context.Context,
 				err)
 		}
 
+		//nolint:lll
 		fundingReq, err := b.cfg.RequestParser(&lnrpc.OpenChannelRequest{
-			SatPerVbyte:        uint64(req.SatPerVbyte),
-			NodePubkey:         rpcChannel.NodePubkey,
-			LocalFundingAmount: rpcChannel.LocalFundingAmount,
-			PushSat:            rpcChannel.PushSat,
-			TargetConf:         req.TargetConf,
-			Private:            rpcChannel.Private,
-			MinHtlcMsat:        rpcChannel.MinHtlcMsat,
-			RemoteCsvDelay:     rpcChannel.RemoteCsvDelay,
-			MinConfs:           req.MinConfs,
-			SpendUnconfirmed:   req.SpendUnconfirmed,
-			CloseAddress:       rpcChannel.CloseAddress,
-			CommitmentType:     rpcChannel.CommitmentType,
+			SatPerVbyte:                uint64(req.SatPerVbyte),
+			TargetConf:                 req.TargetConf,
+			MinConfs:                   req.MinConfs,
+			SpendUnconfirmed:           req.SpendUnconfirmed,
+			NodePubkey:                 rpcChannel.NodePubkey,
+			LocalFundingAmount:         rpcChannel.LocalFundingAmount,
+			PushSat:                    rpcChannel.PushSat,
+			Private:                    rpcChannel.Private,
+			MinHtlcMsat:                rpcChannel.MinHtlcMsat,
+			RemoteCsvDelay:             rpcChannel.RemoteCsvDelay,
+			CloseAddress:               rpcChannel.CloseAddress,
+			RemoteMaxValueInFlightMsat: rpcChannel.RemoteMaxValueInFlightMsat,
+			RemoteMaxHtlcs:             rpcChannel.RemoteMaxHtlcs,
+			MaxLocalCsv:                rpcChannel.MaxLocalCsv,
+			CommitmentType:             rpcChannel.CommitmentType,
+			ZeroConf:                   rpcChannel.ZeroConf,
```

Comment:
Just thinking aloud here... I'm wondering if there could be some interaction between zero-conf channels and normal channels that are opened in the same transaction? Probably not, but maybe that's worth an integration test?

Reply:
I added a test in the latest commit of this PR that opens a zero-conf channel within the batch and tests if a payment goes through before the batch tx confirms.